safe/gui/tools/print_report_dialog.py

```diff
@@ -0,0 +1,486 @@
+
+
+class PrintReportDialog(QtGui.QDialog, FORM_CLASS):
+    """Print report dialog for the InaSAFE plugin."""
+
+    def __init__(self, impact_function, iface, dock=None, parent=None):
+        """Constructor for the dialog.
+        :param iface: A Quantum GIS QGisAppInterface instance.
+        :type iface: QGisAppInterface
+
+        :param parent: Parent widget of this dialog
+        :type parent: QWidget
+
+        :param dock: Optional dock widget instance that we can notify of
+            changes to the keywords.
+        :type dock: Dock
+
+        .. versionadded: 4.3.0
+        """
+
+        QtGui.QDialog.__init__(self, parent)
+        self.setupUi(self)
+
+        # additional buttons
+        self.button_print_pdf = QtGui.QPushButton(self.tr('Print as PDF'))
+        self.button_print_pdf.setObjectName('button_print_pdf')
+        self.button_print_pdf.setToolTip(self.tr(
+            'Write report to PDF and open it in default viewer'))
+        self.button_box.addButton(
+            self.button_print_pdf, QtGui.QDialogButtonBox.ActionRole)
+
+        self.button_open_composer = QtGui.QPushButton(
+            self.tr('Open in composer'))
+        self.button_open_composer.setObjectName('button_open_composer')
+        self.button_open_composer.setToolTip(
+            self.tr('Prepare report and open it in QGIS composer'))
+        self.button_box.addButton(
+            self.button_open_composer, QtGui.QDialogButtonBox.ActionRole)
+
+        self.button_print_pdf.clicked.connect(self.accept)
+        self.button_open_composer.clicked.connect(self.accept)
+
+        # Save reference to the QGIS interface and parent
+        self.iface = iface
+        self.parent = parent
+        self.dock = dock
+        self.impact_function = impact_function
+
+        self.create_pdf = False
+
+        self.default_template_radio.toggled.connect(
+            self.toggle_template_selectors)
+
+        # Set up things for context help
+        self.help_button = self.button_box.button(QtGui.QDialogButtonBox.Help)
+        # Allow toggling the help button
+        self.help_button.setCheckable(True)
+        self.help_button.toggled.connect(self.help_toggled)
+        self.main_stacked_widget.setCurrentIndex(1)
+
+        self.unwanted_templates = ['merged_report.qpt', 'infographic.qpt']
+
+        # Load templates from resources...
+        template_dir_path = resources_path('qgis-composer-templates')
+        self.populate_template_combobox(
+            template_dir_path, self.unwanted_templates)
+
+        #  ...and user directory
+        settings = QtCore.QSettings()
+        path = settings.value('inasafe/reportTemplatePath', '', type=str)
+
+        if not path:
+            path = join(QgsApplication.qgisSettingsDirPath(), 'inasafe')
+
+        if exists(path):
+            self.populate_template_combobox(path)
+
+        self.restore_state()
+
+    def populate_template_combobox(self, path, unwanted_templates=[]):
```

Comment:
Do not use mutable object (i.e. list) for default parameter. See http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments